### ISSUE TITLE
Behaviour when two lines are numerically similar

### ISSUE THREAD

**Issue by wkerzendorf** [tardis-sn/tardis#1307]
The current behaviour is likely sub-optimal as there is a comparison between the last line interacted and the current line and likely leads to a slightly reduced opacity. This should be investigated and fixed.

**Comment by andrewfullard** [comment]
A diagram of the close line process. Current issues include: if the distance to the line is set to 0, it will always override the electron scattering/boundary interaction case; it's unclear if the total optical depth is being summed appropriately if the packet does not interact with the first line of a close line group

![image](https://user-images.githubusercontent.com/20341051/114764921-85951e00-9d32-11eb-91e4-669cd5a0644e.png)

Note that tau_event is not recalculated after a failed interaction with nu_1. The distance_electron is recalculated using tau_event - tau_total.

However, even if tau_event == tau_total  we had to add an additional check so that the packet will skip electron or boundary interactions if distance_line = 0, i.e. if there is a close line. If distance_line == 0 and tau_event is not recalculated, the close line interaction cannot happen because tau_total will not be increased, and so it cannot become larger than tau_event.

Therefore all that happens with a close line is an extra update to the line estimators.

To summarise, if a packet does not interact with the line at nu_1 in the above image, it will also not interact with nu_2. If nu_3 and nu_4 are also close to nu_1 or nu_2, they will be skipped too

A better option may be to pre-group close lines in the line list at one mean frequency, and check for such groups upon a line interaction; if an interaction occurs, select one of the close lines at random. This is similar to the concept of "superlevels" in Fe I think. The main risk I see is that these close lines are not guaranteed to be from the same elements.

**Comment by andrewfullard** [comment]
Set packet trace moved variable False at start of trace

Check if packet trace moved variable is False at start of line loop

    Check the next line
    If close => distance next line = 0
    Add tau of line
    Repeat until distance line > 0 then trace moved = True
    OR tau event reached (interact)

**Comment by andrewfullard** [comment]
Addendum for boundaries:

If shell incremented, distance to inner boundary infinity
If shell decremented, distance to outer boundary infinity
